File: 프로그래머스/lv1/장정윤/추억 점수.py
# https://school.programmers.co.kr/learn/courses/30/lessons/176963
# 10분
def solution(name, yearning, photo):
    answer = []
    match = {}
    # name, yearing으로 dic 생성
    # photo[idx] 매칭 되는 합산 리턴
    
    for idx, word in enumerate(name):
        match[word] = yearning[idx]
        
    for item_list in photo:
        cnt = 0
        for idx, item in enumerate(item_list):
            if item in match.keys() and match[item]:
                cnt += match[item]
        answer.append(cnt)
        
    return answer

File: 프로그래머스/lv1/leehandsub/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string,int> m;
    
    for(int i=0;i<name.size();i++)
    {
        m.insert({name[i],yearning[i]});
    }
    for(int i=0;i<photo.size();i++)
    {
        int num=0;
        for(int j=0;j<photo[i].size();j++)
        {
            if(m.count(photo[i][j])==1)
            {
                num+=m[photo[i][j]];
            }
        }
        answer.push_back(num);
    }
    
    
    
    return answer;
}
